actions/checkForAndPlaceOrder.ts

```diff
@@ -119,91 +167,80 @@ const _checkForAndPlaceOrder: ActionFn = async (
+  chainId: SupportedChainId,
   conditionalOrder: ConditionalOrder,
+  blockTimestamp: number,
+  blockNumber: number,
   contract: ComposableCoW,
   multicall: Multicall3,
-  chainContext: ChainContext,
-  context: Context
-): Promise<{ deleteConditionalOrder: boolean; error: boolean }> {
-  let error = false;
+  chainContext: ChainContext
+): Promise<PollResultErrors | undefined> {
   try {
-    // Do a basic auth check (for singleOrders) // TODO: Check also Merkle auth
-    // Check in case the user invalidated it (this reduces errors in logs)
-    if (!conditionalOrder.proof) {
-      const ctx = await contract.callStatic.hash(conditionalOrder.params);
-      const authorised = await contract.callStatic
-        .singleOrders(owner, ctx)
-        .catch((error) => {
-          console.log(
-            "[processConditionalOrder] Error checking singleOrders auth",
-            { owner, ctx, conditionalOrder },
-            error
-          );
-          return undefined; // returns undefined if it cannot be checked
-        });
-
-      // Return early if the order is not authorised (if its not authorised)
-      // Note we continue in case of an error (this is just to let _getTradeableOrderWithSignature handle the error and log the Tenderly simulation link)
-      if (authorised === false) {
-        console.log(
-          `[processConditionalOrder] Single order not authed. Deleting order...`,
-          { owner, ctx, conditionalOrder }
-        );
-        return { deleteConditionalOrder: true, error: false };
-      }
-    }
-
-    // Do custom Conditional Order checks
-    const [handler, salt, staticInput] = await (() => {
-      const { handler, salt, staticInput } = conditionalOrder.params;
-      return Promise.all([handler, salt, staticInput]);
-    })();
-    const validateResult = await validateOrder({
+    // TODO: Fix model and delete the explicit cast: https://github.com/cowprotocol/tenderly-watch-tower/issues/18
+    const [handler, salt, staticInput] = conditionalOrder.params as any as [
+      string,
+      string,
+      string
+    ];
+
+    const pollParams = {
+      owner,
+      chainId,
+      blockInfo: {
+        blockTimestamp,
+        blockNumber,
+      },
+      provider: chainContext.provider,
+    };
+    const conditionalOrderParams = {
       handler,
-      salt,
       staticInput,
-    });
-    if (validateResult && validateResult.result !== ValidationResult.Success) {
-      const { result, deleteConditionalOrder } = validateResult;
-      return {
-        error: result !== ValidationResult.FailedButIsExpected, // If we expected the call to fail, then we don't consider it an error
-        deleteConditionalOrder,
-      };
-    }
-
-    // Get GPv2 Order
-    const tradeableOrderResult = await _getTradeableOrderWithSignature(
-      owner,
-      network,
-      conditionalOrder,
-      contract,
-      multicall
+      salt,
+    };
+    let pollResult = await pollConditionalOrder(
```

Comment:
Maybe want some way to pass through offChainInput / proofs.

Reply:
Good point